### PR TITLE
Add missing license headers and add a lint

### PR DESCRIPTION
**Pull request by tbg**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/6113)

### CODE REVIEW

**Comment by mberhault** [comment]
LGTM

File: Makefile [hunk 1]
```diff
@@ -185,6 +185,8 @@ dupl:
 
 .PHONY: check
 check:
+	@echo "checking for missing license headers"
+	@! git grep -LE '^// (Copyright|Code generated by)' '*.go'
```
Comment (tamird) [inline Makefile]: why deviate from the convention of separating arguments from file names with `--`?

File: cmd/protoc-gen-gogoroach/main.go [hunk 2]
```diff
@@ -1,3 +1,19 @@
+// Copyright 2014 The Cockroach Authors.
```
Comment (tamird) [inline cmd/protoc-gen-gogoroach/main.go]: probably 2015

**Comment by tamird** [comment]
LGTM mod comments

**Comment by bdarnell** [comment]
LGTM

File: Makefile (hunk 1 above)
Comment (tschottdorf) [inline Makefile]: No reason, done.

File: cmd/protoc-gen-gogoroach/main.go (hunk 2 above)
Comment (tschottdorf) [inline cmd/protoc-gen-gogoroach/main.go]: Done.

**Pull request merged**